Lib/multiprocessing/context.py

```diff
@@ -199,14 +202,23 @@ def get_start_method(self, allow_none=False):
     def set_start_method(self, method, force=False):
         raise ValueError('cannot set start method of concrete context')
 
-    @property
-    def reducer(self):
+    def get_reducer(self):
         '''Controls how objects will be reduced to a form that can be
         shared with other processes.'''
         return globals().get('reduction')
 
-    @reducer.setter
-    def reducer(self, reduction):
+    def set_reducer(self, reduction):
+        if reduction is original_reducer:
+            globals()['reduction'] = reduction
+            return
+
+        if not isinstance(reduction, original_reducer.AbstractReducer):
+            raise TypeError("Custom reducers must subclass "
+                            "multiprocessing.reduction.AbstractReducer")
+        if not isinstance(reduction.get_pickler_class(), pickle.Pickler):
+            raise TypeError("Custom reducers must return a subclass of "
+                            "pickler.Pickler in the get_pickler_class() method")
+
         globals()['reduction'] = reduction
```

Comment:
I suspect this is not what we want if this is not the default context... (see comment about testing below)